gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/ResumeDagProc.java

```diff
@@ -90,7 +90,7 @@ protected void act(DagManagementStateStore dagManagementStateStore, Optional<Dag
     dagManagementStateStore.checkpointDag(failedDag.get());
 
     // if it fails here, it will check point the failed dag in the (running) dag store again, which is idempotent
-    dagManagementStateStore.deleteFailedDag(failedDag.get());
+    dagManagementStateStore.deleteFailedDag(getDagId());
```

Comment:
I think we should expect this kind of symmetry. It is not like a rest resource, where we do `get resource` and `delete resource`.
It is a table with <dagId, dag>. Consider it like a auto incrementing key `Map<K, V>`, we do not call delete<V> just because we do add(V).
There are many other examples like, `Stack`'s API. 
Basically, I believe, a `key` (dagId/dagNodeId) is a best way to choose a resource/row/dag in this case, not the dag.